### PR TITLE
Refactor and add more event API docker tests

### LINKED ISSUE #779: Events from `block-metadata` mqtt topics don't always relate to block metadata
Subscribing to mqtt topics `block-metadata/accepted` & `block-metadata/confirmed` sometimes return information of `pending` block states twice, instead of a normal behaviour `pending` -> `confirmed`. But fetching from the `client.getBlockMetadata(blockId)` method from the sdk returns the block state correctly.
double-pending_720
pending-confirmed

### LINKED ISSUE #797: Add "visualizer testcase" to the event API docker network test
Add a testcase for #779 after #751 was merged.

### PR DESCRIPTION
**Pull request by jkrvivian**
* Create EventAPIDockerTestFramework that wraps the DockerTestFramework with event API testing variables. 
* Test Transaction Metadata topics 
* Test Transaction with Tagged Data Payload by tag
* Fix #779 by exposing the accepted flag in BlockMetadata
* Test, if accepted/confirmed blocks received from event API, are indeed accepted/confirmed, close #797

### CODE REVIEW

File: pkg/retainer/blockretainer/block_retainer.go [hunk 1]
```diff
@@ -124,11 +124,6 @@ func (r *BlockRetainer) BlockMetadata(blockID iotago.BlockID) (*api.BlockMetadat
 		return nil, ierrors.Wrapf(err, "block %s not found", blockID)
 	}
 
-	// we do not expose accepted flag
```
Comment (muXxer) [inline pkg/retainer/blockretainer/block_retainer.go:127]: do you know why we did that in the first place?
maybe @daria305 remembers?

File: tools/docker-network/tests/eventapiframework.go [hunk 2]
```diff
@@ -0,0 +1,617 @@
...
+	}()
+}
+
+func (e *EventAPIDockerTestFramework) AssertAnchorOutput(ctx context.Context, eventClt *nodeclient.EventAPIClient, anchorId iotago.AnchorID) {
+	outputMetadataChan, subInfo := eventClt.OutputsWithMetadataByAnchorID(anchorId)
+	require.Nil(e.Testing, subInfo.Error())
+
+	go func() {
+		defer subInfo.Close()
+		e.assertOutputMetadataTopics(ctx, "AssertNFTOutput", outputMetadataChan, func(resp *api.OutputWithMetadataResponse) bool {
```
Comment (muXxer) [inline tools/docker-network/tests/eventapiframework.go]: wrong comment

File: tools/docker-network/tests/eventapiframework.go [hunk 3]
```diff
@@ -0,0 +1,617 @@
...
+func (e *EventAPIDockerTestFramework) AwaitEventAPITopics(t *testing.T, cancleFunc context.CancelFunc, numOfTopics int) error {
+	counter := 0
+	timer := time.NewTimer(e.optsWaitFor)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-timer.C:
+			cancleFunc()
+			return ierrors.New("Timeout, did not receive signals from all  topics")
```
Comment (muXxer) [inline tools/docker-network/tests/eventapiframework.go]: two white spaces
```suggestion
			return ierrors.New("Timeout, did not receive signals from all topics")
```

File: tools/docker-network/tests/eventapi_test.go [hunk 4]
```diff
@@ -415,19 +385,51 @@ func test_NFTTransactionBlocks(t *testing.T, d *DockerTestFramework) {
...
+
+		// wait until all topics starts listening
+		err := e.AwaitEventAPITopics(t, cancel, totalTopics)
 		require.NoError(t, err)
+
+		// issue blocks
+		e.SubmitDataBlockStream(account, 5*time.Minute)
+
+		// time's up, cancel the context
+		time.Sleep(5 * time.Minute)
```
Comment (muXxer) [inline tools/docker-network/tests/eventapi_test.go]: why do we wait here?
Shouldn't the `SubmitDataBlockStream` return earliest after the 5 minutes anyway?

**Review by muXxer — CHANGES_REQUESTED** [review]

File: tools/docker-network/tests/eventapi_test.go (hunk 4 above)
Comment (jkrvivian) [reply tools/docker-network/tests/eventapi_test.go]: Ah, this is for debugging, to let the network run a bit longer so I can check with core API. I'll remove it.

**Review by muXxer — APPROVED** [review]

**Pull request merged**